packages/library-base/src/commonMain/kotlin/io/realm/internal/Observable.kt

```diff
@@ -33,11 +33,33 @@ import kotlinx.coroutines.flow.Flow
 //  - Public Observable
 //    fun observe(): Flow<T>
 interface Observable<T> {
-    fun freeze(frozenRealm: RealmReference): Observable<T>?
-    fun thaw(liveRealm: RealmReference): Observable<T>?
     fun registerForNotification(callback: Callback): NativePointer
+
     // FIXME Needs elaborate doc on how to signal and close channel
-    fun emitFrozenUpdate(frozenRealm: RealmReference, change: NativePointer, channel: SendChannel<T>): ChannelResult<Unit>?
-    // Should we have a similar public variant
-    fun observe(): Flow<T>
+    fun emitFrozenUpdate(
+        frozenRealm: RealmReference,
+        change: NativePointer,
+        channel: SendChannel<T>
+    ): ChannelResult<Unit>?
+}
+
+/**
+ * TODO : query
+ */
+interface Freezable<T> {
+    fun freeze(frozenRealm: RealmReference): Observable<T>?
+}
+
+/**
+ * TODO : query
+ */
+interface Thawable<T> {
```

Comment:
Do we really want two interfaces for this? Doesn't freeze and thaw always come in pairs?

Reply:
No, they don't come in pairs.

This is related to how we subscribe to notifications. We can only register for notifications using a results object. This means that when we call `query.asFlow()` the notifier will "thaw" the query, which makes no sense in and of itself, so instead, we create a live results object from the query itself which is then used to register for notifications. This means that a query needs only to be `Thawable` and not `Freezable`, since the output of `query.thaw()` is a `RealmResults` object and that object will be the one that will be "frozen", "thawed" and whatnot.